osu.Game/Database/RealmLive.cs

```diff
@@ -24,14 +24,22 @@ public class RealmLive<T> : ILive<T> where T : RealmObject, IHasGuidPrimaryKey
         /// </summary>
         private readonly T data;
 
+        private readonly RealmContextFactory? realmFactory;
+
         /// <summary>
         /// Construct a new instance of live realm data.
         /// </summary>
         /// <param name="data">The realm data.</param>
-        public RealmLive(T data)
+        /// <param name="realmFactory">The realm factory the data was sourced from. May be null for an unmanaged object.</param>
```

Comment:
I don't feel super sure about this null allowance. Feels like this class is now really two classes at once, because what operations are permissible on the live depends both on the data as well as whether the data was accompanied by a realm factory...

I would actually prefer to have something like the following:
* `RealmLive` always expects to receive a managed object and a realm factory
* `RealmLiveUnmanaged<T> : ILive<T>` is a separate new class which doesn't take a realm factory and only supports reads
* `.ToLiveUnmanaged()` (or `.ToReadOnlyLive()`?) only ever returns the latter, while `.ToLive()` returns the appropriate of the two implementations above depending on value of `IsManaged`

The above works as long as it is accepted that an unmanaged object should never get `.ToLive()`d first and only then added to a realm later and start magically working with writes from that point on.

Seems like the interface should give enough leeway to separate the two paths without peppering too many checks everywhere. In a perfect world the unmanaged impl would only implement a `IReadOnlyLive<T>` to avoid having writes throw at runtime but that is probably too much of a muchness at this point.

Reply:
i'll give this a try. i was considering the same direction but didn't as it's a more time consuming implementation.

Reply:
Have applied these changes, seems good.